Source/WebCore/css/CSSTimingFunctionValue.cpp

```diff
@@ -26,10 +26,34 @@
 #include "config.h"
 #include "CSSTimingFunctionValue.h"
 
+#include <wtf/text/StringBuilder.h>
 #include <wtf/text/StringConcatenateNumbers.h>
 
 namespace WebCore {
 
+String CSSLinearTimingFunctionValue::customCSSText() const
+{
+    if (m_points.isEmpty())
+        return "linear"_s;
+
+    StringBuilder builder;
+    builder.append("linear(");
+    for (size_t i = 0; i < m_points.size(); ++i) {
+        if (i)
+            builder.append(", ");
+
+        const auto& point = m_points[i];
+        builder.append(FormattedNumber::fixedPrecision(point.value), ' ', FormattedNumber::fixedPrecision(point.progress * 100.0), '%');
+    }
+    builder.append(')');
+    return builder.toString();
+}
+
+bool CSSLinearTimingFunctionValue::equals(const CSSLinearTimingFunctionValue& other) const
+{
+    return m_points == other.m_points;
```

Comment:
Does this actually check that the two `Vector<LinearTimingFunction::Point>`s have the same values? I guess I'm asking what `LinearTimingFunction::Point` is.

Reply:
`LinearTimingFunction::Point` is basically a named tuple of `double value` (i.e. what the transformed/animated value should be) and `double progress` (i.e. at what point in the transition/animation).  it has it's own `operator==` so it should be using that to compare